synapse/module_api/__init__.py

```diff
@@ -1520,6 +1521,26 @@ async def get_monthly_active_users_by_service(
             start_timestamp, end_timestamp
         )
 
+    async def get_room_alias(self, room_id: RoomID) -> Optional[RoomAlias]:
+        """
+        Get the room alias associated with a room ID.
+
+        Added in Synapse v1.82.0.
+
+        Args:
+            room_id: The Room ID to find the alias of.
+
+        Returns:
+            The optional room alias
+        """
+        from synapse.api.constants import EventTypes
+
+        alias_event = await self._storage_controllers.state.get_current_state_event(
+            room_id, EventTypes.CanonicalAlias, ""
+        )
```

Comment:
Probably want to use this function, since it already exists:

https://github.com/matrix-org/synapse/blob/d0c713cc85f094c323b2ba3f02d8ac411a7f0705/synapse/storage/controllers/state.py#L445-L467

Though it looks like that function has a bug: it looks up the `canonical_alias` field in the event content, not `alias` as [specced](https://spec.matrix.org/v1.6/client-server-api/#mroomcanonical_alias).  Could you fix that up while we're here?

Reply:
(I think) I just fixed this, did you mean reading the field in the event at the very end of the function?